mrtt-ui/src/components/SiteInterventions.js

```diff
@@ -84,29 +101,67 @@ function SiteInterventionsForm() {
     remove: biophysicalInterventionsRemove
   } = useFieldArray({ name: 'biophysicalInterventionsUsed', control })
 
+  const {
+    fields: mangroveSpeciesUsedFields,
+    append: mangroveSpeciesUsedAppend,
+    remove: mangroveSpeciesUsedRemove,
+    update: mangroveSpeciesUsedUpdate
+  } = useFieldArray({ name: 'mangroveSpeciesUsed', control })
+
   const { siteId } = useParams()
   const apiAnswersUrl = `${process.env.REACT_APP_API_URL}/sites/${siteId}/registration_answers`
   const [isSubmitting, setIsSubmitting] = useState(false)
   const [isSubmitError, setIsSubmitError] = useState(false)
   const [isLoading, setIsLoading] = useState(false)
   const [biophysicalInterventionTypesChecked, setBiophysicalInterventionTypesChecked] = useState([])
+  const [mangroveSpeciesForCountriesSelected, setMangroveSpeciesForCountriesSelected] = useState([])
+  const [mangroveSpeciesUsedChecked, setMangroveSpeciesUsedChecked] = useState([])
   const localParticipantTrainingWatcher = watchForm('localParticipantTraining')
 
-  const setInitialBiophysicalInterventionTypesFromServerData = useCallback((serverResponse) => {
+  const loadServerData = useCallback((serverResponse) => {
+    // set biophysical interventions
     const biophysicalInterventionsInitialVal = getBiophysicalInterventions(serverResponse)
 
     const initialBiophysicalInterventionsTypesChecked = biophysicalInterventionsInitialVal?.map(
       (intervention) => intervention.interventionType
     )
     setBiophysicalInterventionTypesChecked(initialBiophysicalInterventionsTypesChecked)
+
+    // set countries list for countries selected in 1.2
+    const siteCountriesResponse = getSiteCountries(serverResponse)
+
+    if (siteCountriesResponse.length) {
+      const countriesList = siteCountriesResponse.map(
+        (countryItem) => countryItem.properties.country
+      )
+      const species = []
+      countriesList.forEach((countrySelected) => {
+        mangroveSpeciesPerCountryList.forEach((countryItem) => {
+          if (countryItem.country.name === countrySelected) {
+            species.push(...countryItem.species)
+          }
+        })
+      })
+      const uniqueSpecies = [...new Set(species)]
+
+      setMangroveSpeciesForCountriesSelected(uniqueSpecies)
+    }
+
+    // set mangrove species list for items selected in 6.2b
+    const mangroveSpeciesUsedResponse = getMangroveSpeciesUsed(serverResponse)
```

Comment:
instead of comments you could just make the function `getMangroveSpeciesUsedFrom6_2b`, etc.

Reply:
good idea! will rename